src/sentry/sentry_metrics/querying/metadata/metrics_code_locations.py

```diff
@@ -148,8 +153,12 @@ def _get_code_locations(
             parsed_locations = [
                 self._parse_code_location_payload(location) for location in locations
             ]
-            # To maintain consistent ordering, we sort by filename.
-            sorted_locations = sorted(parsed_locations, key=lambda value: value.filename or "")
+            # To maintain consistent ordering, we sort by filename, and we want to only get half of the maximum set
+            # size. This way, we will have consistent ordering up to MAX_SET_SIZE elements in the set, if the size
+            # goes above that, we will end up having different locations returned each time (due to the random access).
+            sorted_locations = sorted(parsed_locations, key=lambda value: value.filename or "")[
```

Comment:
It's likely that the locations are all from the same file, maybe should sort by `(filename, line)`?